test/common/common/perf_annotation_test.cc

```diff
@@ -26,6 +26,10 @@ TEST_F(PerfAnnotationTest, TestMacros) {
   PERF_RECORD(perf, "beta", "1");
   PERF_RECORD(perf, "alpha", "2");
   PERF_RECORD(perf, "beta", "3");
+  { PERF_BEGIN("gamma", "4"); }
+  { PERF_BEGIN("sigma", "5"); }
+  { PERF_END("gamma", "4"); }
+  { PERF_END("sigma", "5"); }
```

Comment:
Can you also add begin/end checking with same-category different-description and vice versa?